app/controllers/profiles_controller.rb

```diff
@@ -13,13 +13,9 @@ def update
   end
 
   def search
-    @results = []
-    Profile.all.each do |profile|
-      if (@profile.distance_to(profile.postcode) <= params[:distance].to_f) && (profile != @profile)
-        @results << [profile, @profile.distance_to(profile.postcode).round(2)]
-      end
-    end 
-     @results.sort!{|a, b| a[1] <=> b[1]}
+    @results = Profile.near(@profile.postcode, params[:distance].to_f).to_a
```

Comment:
Hi Dan,

This suggested method doesn't seem to work for us.  We're running into
trouble with the 'where' statement which we can't get to return us anything
(using Rails c).. even when we're searching for a simple field like
"name"... maybe our syntax is incorrect?

*Tamlyn Rudolph*

T:    0439 544 034
E:   tamlyn@ethicalswitch.com
W:  ethicalswitch.com




On Wed, Jul 20, 2016 at 1:42 PM, Dan Garland <notifications@github.com>
wrote:

> In app/controllers/profiles_controller.rb
> <https://github.com/wegotcoders/wegotsingles/pull/10#discussion_r71516643>
> :
>
> > @@ -13,13 +13,9 @@ def update
> >    end
> >
> >    def search
> > -    @results = []
> > -    Profile.all.each do |profile|
> > -      if (@profile.distance_to(profile.postcode) <= params[:distance].to_f) && (profile != @profile)
> > -        @results << [profile, @profile.distance_to(profile.postcode).round(2)]
> > -      end
> > -    end
> > -     @results.sort!{|a, b| a[1] <=> b[1]}
> > +    @results = Profile.near(@profile.postcode, params[:distance].to_f).to_a
>
> Just noticed you have a customer id, how about
> Profile.near(@profile.postcode, params[:distance].to_f).where("customer_id
> != ?" current_customer.id) ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wegotcoders/wegotsingles/pull/10/files/7f5b846474e4d8a4de106c2b25cd5ae40dbbb361..8ec3c462d89c44b08c413ca5e6773335074d2240#r71516643>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AP6wSyLfvh6nxGH2Tk2MflvLSY1kF7E-ks5qXhfMgaJpZM4JQNpb>
> .
>